File: string/atoi.js
// https://leetcode.com/problems/string-to-integer-atoi/
/**
 * @param {string} str
 * @return {number}
 */
const myAtoi = (str) => {
  const match = str.match(/^ *([+-]{0,1}[0-9]+)/);
  return match ? Math.min(2147483647, Math.max(-2147483648, match[1])) : 0;
};

File: string/atoi.py
# https://leetcode.com/problems/string-to-integer-atoi/
import re


class Solution:
    def myAtoi(self, str: str) -> int:
        str = str.strip()
        if not str:
            return 0
        maxint = 2147483647
        minint = -2147483648
        exp = r'[-+]?\d+'
        str = re.match(exp, str)
        if str:
            num = int(str.group())
            if num > maxint:
                return maxint
            elif num < minint:
                return minint
            else:
                return num
        return 0
